src/lazy/any_encoding.rs

```diff
@@ -402,6 +396,16 @@ impl<'data> LazyRawReader<'data, AnyEncoding> for LazyRawAnyReader<'data> {
             Binary_1_1(r) => r.position(),
         }
     }
+
+    fn encoding(&self) -> IonEncoding {
```

Comment:
🗺️ All raw readers now provide a `fn encoding(&self) -> IonEncoding`, which allows generic code to tell which Ion encoding it's working with.